save-demo/src/main/kotlin/com/saveourtool/save/demo/runners/pod/PodRunner.kt

```diff
@@ -28,35 +29,24 @@ interface PodRunner : Runner {
      */
     @Suppress("InjectDispatcher")
     override fun run(demoRunRequest: DemoRunRequest): Mono<DemoResult> = Mono.just(demoRunRequest)
-        .doOnNext { logger.debug("Getting url for demo execution.") }
-        .flatMap { getUrl(demoRunRequest) }
         .doOnNext { logger.info("Sending run request to $it.") }
-        .flatMap { mono(Dispatchers.Default) { runRequest(it, demoRunRequest) } }
+        .flatMap { mono { runRequest(it) } }
+        .switchIfEmptyToResponseException(HttpStatus.INTERNAL_SERVER_ERROR) {
+            "Could not fetch demo result."
+        }
         .doOnNext { logger.info("Demo finished with [${it.terminationCode}] code.") }
 
-    private suspend fun runRequest(
-        url: String,
-        demoRunRequest: DemoRunRequest,
-    ): DemoResult? = httpClient.post(url) {
-        contentType(ContentType.Application.Json)
-        accept(ContentType.Application.Json)
-        setBody(demoRunRequest)
+    private suspend fun runRequest(demoRunRequest: DemoRunRequest): DemoResult? = sendRunRequest(demoRunRequest).let { response ->
+        response?.takeIf { it.status.isSuccess() }?.body()
     }
-        .let { response -> response.takeIf { it.status.isSuccess() }?.body() }
 
     /**
-     * @param demoRunRequest
-     * @return url of a pod that runs demo
+     * @param demoRunRequest run request
+     * @return [HttpResponse] if response received, null if something went wrong and response was not received
      */
-    fun getUrl(demoRunRequest: DemoRunRequest): Mono<String>
+    suspend fun sendRunRequest(demoRunRequest: DemoRunRequest): HttpResponse?
```

Comment:
will it better to create a manual http response instead of null, in case of some errors?

Reply:
Didn't get you point

Reply:
Discussed offline - return nullable is ok as far as it is processed on `Controller` level - wrapped into `Mono` and checked to be empty